src/Editor.js

```diff
@@ -0,0 +1,126 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { Controlled as CodeMirror } from 'react-codemirror2';
+
+import 'codemirror/lib/codemirror.css';
+import 'codemirror/theme/solarized.css';
+import 'codemirror/mode/python/python';
+import 'codemirror/mode/clike/clike';
+import 'codemirror/mode/php/php';
+import 'codemirror/mode/ruby/ruby';
+import 'codemirror/mode/javascript/javascript';
+import 'codemirror/mode/go/go';
+import 'codemirror/mode/shell/shell';
+
+import splitProps from './splitProps';
+
+class Editor extends Component {
+  constructor(props) {
+    super(props);
+
+    this.bookmark = null;
+  }
+
+  get document() {
+    return this.editor.getDoc();
+  }
+
+  get editor() {
+    return this.codemirror.editor;
+  }
+
+  selectCode(range) {
+    if (this.mark) {
+      this.mark.clear();
+    }
+
+    if (!range) {
+      this.mark = null;
+      return;
+    }
+
+    const { from, to } = range;
+
+    if (from && to) {
+      this.mark = this.document.markText(from, to, {
+        className: 'uast-editor-mark'
+      });
+    } else if (from && this.bookmark) {
+      this.mark = this.document.setBookmark(from, {
+        widget: this.bookmark
+      });
+    }
+  }
+
+  componentDidMount() {
+    this.bookmark.remove();
+
+    this.selectCode(this.props.markRange);
+  }
+
+  componentDidUpdate(prevProps) {
+    this.bookmark.remove();
+
+    if (prevProps.markRange !== this.props.markRange) {
+      this.selectCode(this.props.markRange);
+    }
+  }
+
+  onCursor(pos) {
+    if (!this.props.onCursorChanged) {
+      return;
+    }
+
+    this.props.onCursorChanged(pos);
+  }
+
+  render() {
+    const [props, childProps] = splitProps(this.props, Editor);
+    const { code, languageMode, onChange } = props;
+
+    const options = {
+      mode: languageMode,
+      lineNumbers: true,
+      theme: 'solarized light'
```

Comment:
Maybe the theme should be open to be configured via props, with this one as default.